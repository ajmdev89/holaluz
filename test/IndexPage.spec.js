import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuex, { Store } from 'vuex'
import Landing from '~/pages/index.vue'

describe('OffersData component -----OffersData.vue-----', () => {
  let wrapper
  let store

  Vue.use(Vuex)

  beforeEach(() => {
    store = new Store({
      actions: {
        getOffers: jest.fn()
      }
    })

    wrapper = shallowMount(Landing, {
      store
    })
  })

  it('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('it should render', () => {
    expect(wrapper.find('h1').exists()).toBeTruthy()
    expect(wrapper.find('.bg-gradient-to-r').exists()).toBeTruthy()
  })

  it('it should render without offer section', () => {
    expect(wrapper.find('.bg-white').exists()).toBeFalsy()
  })
})
