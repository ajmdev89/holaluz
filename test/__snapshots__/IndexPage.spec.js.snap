// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersData component -----OffersData.vue----- matches snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
